DirectProgramming/DPC++FPGA/include/cached_local_memory.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef __CACHED_LOCAL_MEMORY_HPP__
+#define __CACHED_LOCAL_MEMORY_HPP__
+
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+
+#include "constexpr_math.hpp"   // DirectProgramming/DPC++FPGA/include
+
+namespace fpga_tools {
+
+template <typename T,             // type to store in the memory
+          size_t k_mem_depth,     // depth of the memory
```

Comment:
Looking at the other templated functions, I don't think that we prefix our integer template parameters with k_

Reply:
I really like this convention, I use it in all my templated code (including several places in the code samples).

kVariableName is the convention for constants.  Integer template parameters are compile time constants, but someone on the code samples group ages ago objected to using kVariableName format for them.  So this was my compromise to indicate that they are not 'variables', and can be used safely as constants, and will behave as constants in terms of performance.

You're in charge now, so if you want me to change this in the shared headers I will, but as I said I find this a helpful convention.

Reply:
I think that this is a valid point and I'll make a note to update the other code samples to follow that convention